src/pudl/transform/classes.py

```diff
@@ -297,6 +297,47 @@ def enforce_snake_case(
 enforce_snake_case_multicol = multicol_transform_factory(enforce_snake_case)
 
 
+################################################################################
+# Strip Non-Numeric Values
+################################################################################
+class StripNonNumericValues(TransformParams):
+    """Boolean parameter for :func:`strip_non_numeric_values`.
+
+    Stores a named boolean variable that is employed in
+    :func:`strip_non_numeric_values` to determine whether of not the transform
+    treatment should be applied. Pydantic 2.0 will allow validation of these simple
+    variables without needing to define a model.
+    """
+
+    strip_non_numeric_values: bool
+
+
+def strip_non_numeric_values(
+    col: pd.Series, params: StripNonNumericValues | None = None
+) -> pd.Series:
+    """Strip a column of any non-integer values.
+
+    Using the following options in :func:`pd.Series.extract` :
+
+    * an optional ``+`` or ``-`` followed by at least one digit followed by an optional
+      decimal place followed by any number of digits (including zero)
+    * OR an optional ``+`` or ``-`` followed by a period followed by at least one digit
+
+    Note: this will not work with exponential values.
+    """
+    if params is None:
+        params = StripNonNumericValues(strip_non_numeric_values=True)
+    if params.strip_non_numeric_values:
+        col = col.astype(str).str.extract(
+            rf"(?P<{col.name}>[-+]?\d+\.?\d*|[-+]?\.\d+)",  # name the series
+            expand=False,
+        )
+    return col
```

Comment:
In a case where the string is something like "FERC1 Proj. No. 123" would this strip to "1" instead of "123"? Tbh I can't really decipher this regex so maybe this case works, but also could be a nice test case for the unit test. I guess this would require checking that there's not a letter right before a the number?

Reply:
hm yeah i think you may be right that this would strip "1" instead of "123". regex is not my strong suit and the behavior of extract finding the first instance of the regex is tres annoying imo

Reply:
Ya I think making `expand=True` is likely to lead to more errors. I think a check to ensure there's no letter before the number is the best simple thing you can do.

Reply:
done! learned a new regex name: "Negative Lookbehind"